Source/JavaScriptCore/runtime/JSFunction.cpp

```diff
@@ -595,11 +607,11 @@ void JSFunction::reifyName(VM& vm, JSGlobalObject* globalObject, String name)
         if (name.find(illegalCharMatcher) != notFound)
             name = String();
     }
-    
+
     if (jsExecutable()->isGetter())
-        name = makeString("get ", name);
+        makeNameWithOutOfMemoryCheck(globalObject, throwScope, "Getter ", "get ", name);
     else if (jsExecutable()->isSetter())
-        name = makeString("set ", name);
+        makeNameWithOutOfMemoryCheck(globalObject, throwScope, "Setter ", "set ", name);
 
     rareData->setHasReifiedName();
```

Comment:
I just realized that this is wrong.  The patch as you have it would `setHasReifiedName` even though you fail to do so.  You should have `makeNameWithOutOfMemoryCheck` return a bool to indicate success or not.  It not successful, return with `PropertyStatus::Lazy`.  Also, to placate the exception check validator, you'll need to `ASSERT(!throwScope.exception || success)` before the `if (success)` check.

Reply:
You can tests for this scenario by running your test cases twice e.g.
```
shouldThrow(classSetter, setterExpectedError);
shouldThrow(classSetter, setterExpectedError); // Make sure it should still throw.
```
If you didn't fix this bug, I suspect the 2nd test case invocation above till fail.

Reply:
I think this bug is one of the reasons you're getting EWS failures.